tst/EnergyPlus/unit/OutdoorAirUnit.unit.cc

```diff
@@ -358,4 +371,618 @@ TEST_F(EnergyPlusFixture, OutdoorAirUnit_AutoSize)
+
+    UpdateScheduleValues();
+
+    ZoneEqSizing.allocate(1);
+    CurDeadBandOrSetback.allocate(1);
+    CurDeadBandOrSetback(1) = false;
+    TempControlType.allocate(1);
+    TempControlType(1) = 4;
+
+    ZoneSizingRunDone = true;
+    DataSizing::CurZoneEqNum = 1;
+    ZoneEqSizing(CurZoneEqNum).DesignSizeFromParent = false;
+    ZoneEqSizing(CurZoneEqNum).SizingMethod.allocate(25);
+    ZoneEqSizing(CurZoneEqNum).SizingMethod(DataHVACGlobals::SystemAirflowSizing) = DataSizing::SupplyAirFlowRate;
+
+    FinalZoneSizing.allocate(1);
+    FinalZoneSizing(CurZoneEqNum).MinOA = 0.5;
+    FinalZoneSizing(CurZoneEqNum).DesHeatVolFlow = 0.5;
+    FinalZoneSizing(CurZoneEqNum).DesHeatCoilInTemp = 5.0;
+    FinalZoneSizing(CurZoneEqNum).DesHeatCoilInHumRat = 0.005;
+
+    DataEnvironment::StdRhoAir = PsyRhoAirFnPbTdbW(OutBaroPress, 5.0, 0.0);
+    OutAirUnit(OAUnitNum).OAEquip(1).MaxVolWaterFlow = DataSizing::AutoSize;
+
+    FinalZoneSizing(CurZoneEqNum).HeatDesTemp = 50.0;
+    FinalZoneSizing(CurZoneEqNum).HeatDesHumRat = 0.0050;
+    FinalZoneSizing(CurZoneEqNum).DesHeatDens = DataEnvironment::StdRhoAir;
+    FinalZoneSizing(CurZoneEqNum).DesHeatMassFlow = FinalZoneSizing(CurZoneEqNum).DesHeatVolFlow * FinalZoneSizing(CurZoneEqNum).DesHeatDens;
+
+    BeginEnvrnFlag = true;
+    bool FirstHVACIteration(true);
+    int ZoneNum(1);
+
+    InitOutdoorAirUnit(OAUnitNum, ZoneNum, FirstHVACIteration);
+    EXPECT_EQ(SteamCoil(1).MaxSteamVolFlowRate, OutAirUnit(OAUnitNum).OAEquip(1).MaxVolWaterFlow);
+
+    Real64 DesCoilInTemp = FinalZoneSizing(CurZoneEqNum).DesHeatCoilInTemp;
+    Real64 DesCoilOutTemp = FinalZoneSizing(CurZoneEqNum).HeatDesTemp;
+    Real64 DesCoilOutHumRat = FinalZoneSizing(CurZoneEqNum).HeatDesHumRat;
+    Real64 DesAirMassFlow = FinalZoneSizing(CurZoneEqNum).DesHeatMassFlow;
+    // DesVolFlow = DesMassFlow / RhoAirStd;
+    Real64 CpAirAvg = PsyCpAirFnWTdb(DesCoilOutHumRat, 0.5 * (DesCoilInTemp + DesCoilOutTemp));
+    Real64 DesSteamCoilLoad = DesAirMassFlow * CpAirAvg * (DesCoilOutTemp - DesCoilInTemp);
+
+    // do steam flow rate sizing calculation
+    Real64 EnthSteamIn = GetSatEnthalpyRefrig("STEAM", DataGlobals::SteamInitConvTemp, 1.0, SteamCoil(1).FluidIndex, "");
+    Real64 EnthSteamOut = GetSatEnthalpyRefrig("STEAM", DataGlobals::SteamInitConvTemp, 0.0, SteamCoil(1).FluidIndex, "");
+    Real64 SteamDensity = GetSatDensityRefrig("STEAM", DataGlobals::SteamInitConvTemp, 1.0, SteamCoil(1).FluidIndex, "");
+    Real64 CpOfCondensate = GetSatSpecificHeatRefrig("STEAM", DataGlobals::SteamInitConvTemp, 0.0, SteamCoil(1).FluidIndex, "");
+    Real64 LatentHeatChange = EnthSteamIn - EnthSteamOut;
+    Real64 DesMaxSteamVolFlowRate = DesSteamCoilLoad / (SteamDensity * (LatentHeatChange + SteamCoil(1).DegOfSubcooling * CpOfCondensate));
+
+    // check water coil water flow rate calc
+    EXPECT_EQ(DesSteamCoilLoad, SteamCoil(1).DesCoilCapacity);
+    EXPECT_EQ(DesMaxSteamVolFlowRate, SteamCoil(1).MaxSteamVolFlowRate);
```

Comment:
I won't say no to this ( I might tomorrow), but 600 lines of code to prove capacity = load, we have to get better at this,

Reply:
I thought the unit test were intended to demonstrate the fix. This unit test demonstrates just that. Before the fix, steam coil cannot be used in Outdoor Unit and I am demonstrating that and also demonstrating how the steam density is used to calculate steam flow rate.

Reply:
@Nigusse I think @rraustad is just lamenting that it took 600 lines of test code to demonstrate it.  If I'm reading it properly, you did everything just fine.  As we continue to objectify the code, the amount of code required to create nice unit tests **will** reduce, and it will become an easier lift to write more and more tests.